src/main/java/chatbot/Parser.java

```diff
@@ -0,0 +1,184 @@
+import java.util.List;
+
+public class Parser {
+    /**
+     * Parses user input for add method. If input is appropriate, returns an array containing the type of task,
+     * description, and other parameters, in that order.
+     *
+     * @param input The command inputted by the user.
+     * @returns An array containing the type and details of the task.
+     * @throws InvalidTodoException If format of todo entered is invalid.
+     * @throws InvalidDeadlineException If format of deadline entered is invalid.
+     * @throws InvalidEventException If format of event entered is invalid.
+     * @throws InvalidCommandException If format of command entered is invalid.
+     */
+    public List<String> parseAdd(String input) throws InvalidTodoException,
+            InvalidDeadlineException, InvalidEventException, InvalidCommandException {
+        List<String> arr = new ArrayList<>();
+        String nextInput;
+
+        try {
+            nextInput = input.substring(4);
+
+            if (nextInput.isBlank()) {
+                throw new InvalidCommandException();
+            }
+        } catch (StringIndexOutOfBoundsException e) {
+            throw new InvalidCommandException();
+        }
+
+        if (nextInput.startsWith("todo")) {
+            try {
+                String description = nextInput.substring(5);
+
+                if (description.isBlank()) {
+                    throw new InvalidTodoException();
+                } else {
+                    arr.add(0, "todo");
+                    arr.add(1, description);
+                }
+            } catch (StringIndexOutOfBoundsException e) {
+                throw new InvalidTodoException();
+            }
+        } else if (nextInput.startsWith("deadline")) {
+            try {
+                int by = nextInput.indexOf("/by");
+                String description = nextInput.substring(9, by - 1);
+                String deadline = nextInput.substring(by + 4);
+
+                if (description.isBlank() || deadline.isBlank()) {
+                    throw new InvalidDeadlineException();
+                } else {
+                    arr.add(0,"deadline");
+                    arr.add(1, description);
+                    arr.add(2, deadline);
+                }
+            } catch (StringIndexOutOfBoundsException e) {
+                throw new InvalidDeadlineException();
+            }
+        } else if (nextInput.startsWith("event")) {
+            try {
+                int from = nextInput.indexOf("/from");
+                int to = nextInput.indexOf("/to");
+                String description = nextInput.substring(6, from - 1);
+                String start = nextInput.substring(from + 6, to - 1);
+                String end = nextInput.substring(to + 4);
+
+                if (description.isBlank() || start.isBlank() || end.isBlank()) {
+                    throw new InvalidEventException();
+                } else {
+                    arr.add(0, "event");
```

Comment:
I like how u renamed the string inputs into something more human readable, etc description, start and end